numpy/_core/tests/test_regression.py

```diff
@@ -109,7 +109,9 @@ def test_noncontiguous_fill(self):
         def rs():
             b.shape = (10,)
 
-        assert_raises(AttributeError, rs)
+        with warnings.catch_warnings():  # gh-28901
```

Comment:
Since python 3.11, one can just do `with warnings.catch_warnings(action="ignore", category=DeprecationWarning)`.

Though I think just using `pytest.warns(DeprecationWarning)`, as you do below, is cleaner.